app/src/androidTest/java/com/hedvig/app/feature/referrals/tab/ReferralTabShareTest.kt

```diff
@@ -0,0 +1,121 @@
+class ReferralTabShareTest : KoinTest {
+    private val apolloClientWrapper: ApolloClientWrapper by inject()
+
+    @get:Rule
+    val activityRule = IntentsTestRule(LoggedInActivity::class.java, false, false)
+
+    @Before
+    fun setup() {
+        apolloClientWrapper
+            .apolloClient
+            .clearNormalizedCache()
+    }
+
+    @Test
+    fun shouldOpenShareWhenClickingShare() {
+        MockWebServer().use { webServer ->
+            webServer.dispatcher = object : Dispatcher() {
+                override fun dispatch(request: RecordedRequest): MockResponse {
+                    val body = request.body.peek().readUtf8()
+                    if (body.contains("Features")) {
+                        return MockResponse().setBody(FEATURES_DATA.toJson())
+                    }
+
+                    if (body.contains("Referrals")) {
+                        return MockResponse().setBody(REFERRALS_DATA.toJson())
+                    }
+
+                    return MockResponse()
+                }
+            }
+
+            webServer.start(8080)
+
+            val intent = LoggedInActivity.newInstance(
+                ApplicationProvider.getApplicationContext(),
+                initialTab = LoggedInTabs.REFERRALS
+            )
+
+            activityRule.launchActivity(intent)
+
+            intending(not(isInternal())).respondWith(
+                Instrumentation.ActivityResult(
+                    Activity.RESULT_OK,
+                    null
+                )
+            )
+
+            Screen.onScreen<ReferralScreen> {
+                share {
+                    isVisible()
+                    click()
+                }
+                shareIntent { intended() }
+            }
+        }
+    }
+
+    companion object {
+        private val FEATURES_DATA = FeaturesQuery.Data(
+            member = FeaturesQuery.Member(
+                features = listOf(
+                    Feature.KEYGEAR
+                )
+            )
+        )
+
+        private val REFERRALS_DATA = ReferralsQuery.Data(
+            referralInformation = ReferralsQuery.ReferralInformation(
+                campaign = ReferralsQuery.Campaign(
+                    code = "TEST123",
+                    incentive = ReferralsQuery.Incentive(
+                        asMonthlyCostDeduction = ReferralsQuery.AsMonthlyCostDeduction(
+                            amount = ReferralsQuery.Amount(
+                                fragments = ReferralsQuery.Amount.Fragments(
+                                    MonetaryAmountFragment(
+                                        amount = "10.00",
+                                        currency = "SEK"
+                                    )
+                                )
+                            )
+                        )
+                    )
+                ),
+                referredBy = null,
+                invitations = emptyList()
+            )
+        )
+    }
```

Comment:
I can se that you do this in some places.Could be nice to brake it out in `Builders`

Reply:
At this point I prefer to keep the code DAMP over DRY, because it's not entirely clear how to abstract this in a good way

Reply:
@hedvigoscar let me know if you want to discuss it :)

Reply:
I'm of course a bit intrigued now, lets discuss it! When we have time